lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc

```diff
@@ -44,13 +44,13 @@ function dosomething_signup_form($form, &$form_state, $nid, $label = "Sign Up")
   ];
 
   // Check query string for signup source value.
-  if ($source = dosomething_signup_get_query_source()) {
-    $form['source'] = [
-      '#type' => 'hidden',
-      '#value' => $source,
-      '#access' => FALSE,
-    ];
-  }
+  // if ($source_details = dosomething_signup_get_query_source()) {
```

Comment:
Have you tested that a campaign with a source query string properly fills in this field without the hidden form element? I'm thinking that the call to `dosomething_signup_get_query_source` might not work since it'd be running in the form's POST, where the query string would no longer exist.

(If this does work, let's just remove these commented lines!)